test/lsp.py

```diff
@@ -898,18 +899,29 @@ def main(self) -> int:
 
         return min(max(self.test_counter.failed, self.assertion_counter.failed), 127)
 
-    def setup_lsp(self, lsp: JsonRpcProcess, expose_project_root=True):
+    def setup_lsp(
+        self,
+        lsp: JsonRpcProcess,
+        expose_project_root=True,
+        analyze_all_files_in_project=False,
+        project_root_suffix = None
+    ):
         """
         Prepares the solc LSP server by calling `initialize`,
         and `initialized` methods.
         """
+        rootUri = self.project_root_uri
+        if project_root_suffix is not None:
+            rootUri = rootUri + '/' + project_root_suffix
```

Comment:
I find the naming here confusing because `rootUri` and `self.project_root_uri` sound like the same thing and the "suffix" is actually the middle part in paths. Maybe this would be better?
- `rootUri` -> `project_root_uri_with_subdir`
- `project_root_suffix` -> `project_root_subdir`

Reply:
`rootUri` (now `root_uri` due to your suggestion above) is simply a break-out as I could not easily express that inline anymore. It's having this name as that's the precise name of the parameter key that it is being assigned to right away and not used afterwards anymore. 
`project_root_uri_with_subdir` to me is misleading. it is in fact the project root uri, it's just that sometimes I tweak that to point to a child directory, but not always, and what name what that then be? `project_root_uri_or_deeper_inside_uri`? Hmmm....

> project_root_suffix -> project_root_subdir

done

Reply:
> it is in fact the project root uri

But it isn't. You have two vars that can have different values (`.project_root_uri` and `rootUri`) and only one of them can be the root.